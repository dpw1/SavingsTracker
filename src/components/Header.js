import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <ul>
      <li>
        <NavLink activeClassName="is-active" to="/" exact={true}>
          Home
        </NavLink>
      </li>
      <li>
        <NavLink activeClassName="is-active" to="/create">
          New Expense
        </NavLink>
      </li>

      <li>
        <NavLink activeClassName="is-active" to="/settings">
          Settings
        </NavLink>
      </li>
      <li>
        <NavLink activeClassName="is-active" to="/help">
          Help
        </NavLink>
      </li>
    </ul>
  </header>
);

export default Header;
