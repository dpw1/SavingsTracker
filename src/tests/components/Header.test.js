import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Header from "../../components/Header";
import ReactShallowRenderer from "react-test-renderer/shallow";

test("should render Header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
  //   expect(wrapper.find("ul").length).toBe(1);
  //   const renderer = new ReactShallowRenderer();
  //   renderer.render(<Header />);
  //   expect(renderer.getRenderOutput()).toMatchSnapshot();
});
