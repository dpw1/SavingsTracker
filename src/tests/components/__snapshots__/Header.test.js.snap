// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        New Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/settings"
      >
        Settings
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</header>
`;
